src/app/forms/components/date/date.component.html

```diff
@@ -2,7 +2,11 @@
   <fieldset class="govuk-fieldset" role="group" attr.aria-describedby="{{ name }}-hint">
     <legend class="govuk-fieldset__legend govuk-fieldset__legend--m">{{ label }}</legend>
     <div *ngIf="hint" id="{{ name }}-hint" class="govuk-hint">{{ hint }}</div>
-    <app-field-error-message [error]="error" [name]="name"></app-field-error-message>
+    <app-field-error-message
+      *ngIf="dayEl.dirty && monthEl.dirty && yearEl.dirty && yearEl.valid"
```

Comment:
why `yearEl.valid` ? don't we want to show an error in the year is not valid?

Reply:
looks good otherwise

Reply:
isn't just checking all of them are `dirty` enough?

Reply:
No, because as soon as you push '1' into year you get the error message again

Reply:
it really needs to be year dirty but not being touched, not sure how to do that here?

Reply:
makes sense... I personally would say that would be fine, but it's not really up to me...

I that case, I'd say slight nitpick: can we check the length is greater than 3 then ?

Reply:
I like checking year.length > 3 as well yeah! I'll see if I can work out a way to tell if the control is currently being touched

Reply:
I think the sell on this is 'perfection' is probably really hard and just covering off the likely scenario, plus having it validate 100% at submission is probably okay

Reply:
```suggestion
      *ngIf="dayEl.dirty && monthEl.dirty && yearEl.dirty && yearEl.value.length > 3"
```